pkg/models/datasource_cache.go

```diff
@@ -11,17 +11,87 @@ import (
 	"time"
 
 	"github.com/grafana/grafana/pkg/setting"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+var datasourceRequestCounter = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Namespace: "grafana",
+		Name:      "datasource_request_total",
+		Help:      "A counter for outgoing requests for a datasource",
+	},
+	[]string{"datasource", "code", "method"},
```

Comment:
Datasource name as label may be subject for high cardinality. Possibility of same name in multiple organizations. Would using data source type make more sense and/or in addition to datasource name?

Reply:
I think it would be very very very rare that the database count causes high cardinality problems but it could since its unbound in theory. How about we introduce a setting for that if it becomes an issue?

OrgId is a good point... Hm. how about a new orgId label? Cardinality wise is the same since it will always be the same per datasource.

Reply:
> How about we introduce a setting for that if it becomes an issue?

Sounds good. 

> OrgId is a good point... Hm. how about a new orgId label? Cardinality wise is the same since it will always be the same per datasource.

I don't know. Maybe wait with this one as well until it becomes a problem given "How about we introduce a setting for that if it becomes an issue"

Reply:
:+1: I'll revert my commit and merge this.